options/secrets.go

```diff
@@ -0,0 +1,45 @@
+package options
+
+import (
+	"fmt"
+	"os"
+
+	"github.com/coder/serpent"
+)
+
+var buildSecretPrefix = fmt.Sprintf("%sBUILD_SECRET_", envPrefix)
+
+func EnvWithBuildSecretPrefix(secretName, secretValue string) string {
```

Comment:
Suggestion: It's best practice for exported functions to have documentation (`// EnvWithBuildSecretPrefix ...`). It's part of the original Go linter (https://pkg.go.dev/github.com/golang/lint) and effective go, so a bit surprising.

```
❯ golint ./...
envbuilder.go:43:2: a blank import should be only in a main or test package, or have a comment justifying it
devcontainer/devcontainer.go:34:6: exported type Spec should have comment or be unexported
devcontainer/devcontainer.go:50:6: exported type LifecycleScripts should have comment or be unexported
devcontainer/devcontainer.go:57:6: exported type BuildSpec should have comment or be unexported
devcontainer/devcontainer.go:78:1: exported function SubstituteVars should have comment or be unexported
devcontainer/script.go:16:6: exported type LifecycleScript should have comment or be unexported
devcontainer/script.go:21:1: exported method LifecycleScript.IsEmpty should have comment or be unexported
```

It seems `golint` is marked deprecated in `golangci-lint` though, so I guess the alternative linters aren't enforcing this rule. 😔 I always found it slightly annoying but ultimately worth the annoyance 😄.

Reply:
I've documented the function and created an issue for us to support these linting cases in envbuilder.